components/Send/AmountInput.tsx

```diff
@@ -1,75 +1,114 @@
 
 interface AmountInputProps {
   onChange: (value: string) => void;
   value: string;
+  tokens: Token[];
+  defaultToken?: Token;
+  selectedTokenBalance: number;
+  onTokenChange?: (token: Token) => void;
 }
 
-export const AmountInput = ({ onChange, value }: AmountInputProps) => {
-  const [isValidAmount, setIsValidAmount] = useState(true);
+export const AmountInput = ({
+  onChange,
+  value,
+  tokens,
+  defaultToken,
+  selectedTokenBalance,
+  onTokenChange,
+}: AmountInputProps) => {
+  const [isPickerOpen, setIsPickerOpen] = useState(false);
+  const [selectedToken, setSelectedToken] = useState<Token | undefined>(
+    defaultToken || (tokens.length > 0 ? tokens[0] : undefined)
+  );
 
-  // const checkFunds = async () => {
-  //   try {
-  //     const balance = await publicClient.getBalance({
-  //       address: '0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045', // Replace with user address
-  //     });
-  //     const formattedBalance = formatEther(balance);
-  //     console.log('Balance:', formattedBalance);
+  const handleInputChange = (value: string) => {
+    onChange(value.replace(/[^0-9.]/g, ''));
+  };
 
-  //     if (parseFloat(amount) > parseFloat(formattedBalance)) {
-  //       Alert.alert(
-  //         'Insufficient Funds',
-  //         `You have ${formattedBalance} ETH, which is not enough to send ${amount} ETH.`
-  //       );
-  //     } else {
-  //       Alert.alert(
-  //         'Sufficient Funds',
-  //         `You have enough balance (${formattedBalance} ETH) to send ${amount} ETH.`
-  //       );
-  //     }
-  //   } catch (error) {
-  //     console.error('Error:', error);
-  //   }
-  // };
+  const isValidAmount =
+    value !== '' && !isNaN(Number(value)) && Number(value) <= selectedTokenBalance;
 
-  const handleInputChange = (value: string) => {
-    const numericValue = value.replace(/[^0-9.]/g, '');
-    if (numericValue.match(/^\d*\.?\d*$/)) {
-      setIsValidAmount(true);
-      onChange(numericValue);
-    } else {
-      setIsValidAmount(false);
-    }
+  const handleTokenSelect = (token: Token) => {
+    setSelectedToken(token);
+    setIsPickerOpen(false);
+    if (onTokenChange) onTokenChange(token);
   };
 
   return (
-    <View className="mb-4 h-40 rounded-xl bg-content p-4">
-      <View className="flex-row items-center">
+    <View className="mb-4 rounded-xl bg-content p-4">
+      <View className="mb-4 flex-row items-center justify-between">
         <FText className="!text-2xl text-text" bold>
           Amount
         </FText>
+        <TouchableOpacity
+          className="flex-row items-center space-x-2 rounded-xl bg-primary px-4 py-1"
+          onPress={() => setIsPickerOpen(true)}>
+          {selectedToken && tokenIcons[selectedToken.symbol] ? (
+            <Image source={tokenIcons[selectedToken.symbol]} className="mr-2 h-6 w-6" />
+          ) : (
+            <Feather name="cpu" size={24} className="mr-2 text-text" />
```

Comment:
if you plan on using `bg-primary` for the input, then you must always use text-white for its content
```suggestion
            <Feather name="cpu" size={24} className="mr-2 text-white" />
```